types/index.d.ts

```diff
@@ -17,5 +17,5 @@ interface htmlOptions extends HastUtilToHtmlOptions {
   handlers?: Handlers
 }
 
-declare const html: Plugin<[htmlOptions?]>
+declare const html: Plugin<htmlOptions[]>
```

Comment:
The typing as-is is correct.
the `Plugin` type is an alias for `Attacher`
https://github.com/unifiedjs/unified/blob/9b0741b02ccfbd06fc436a7dc22d45f87203547e/types/ts4.0/index.d.ts#L283-L297
which takes a tuple of settings.

As-is it accepts a single, optional `htmlOptions`.
Your change would allow any number of `htmlOptions` which isn't supported.